src/System.Management.Automation/engine/parser/Compiler.cs

```diff
@@ -951,7 +951,7 @@ internal Expression CompileExpressionOperand(ExpressionAst exprAst)
 
         private IEnumerable<Expression> CompileInvocationArguments(IEnumerable<ExpressionAst> arguments)
         {
-            return arguments == null ? Array.Empty<Expression>() : arguments.Select(CompileExpressionOperand);
+            return arguments == null ? Array.Empty<Expression>() : arguments.Select(CompileExpressionOperand).ToArray();
```

Comment:
Looks like everywhere this method is called passes a `ReadOnlyCollection<ExpressionAst>`.  Could type `arguments` as `IReadOnlyList` and do this:

```csharp
if (arguments is null)
{
    return Array.Empty<Expression>();
}

var result = new Expression[arguments.Count];
for (int i = result.Length - 1; i >= 0; i--)
{
    result[i] = CompileExpressionOperand(arguments[i]);
}

return result;
```

If you would rather not change the method parameter type then LGTM!

Reply:
Code updated.